src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,161 @@
+        currentNextNode.previousNode = node;
+        sizeOfList += 1;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node currentNode = getNodeByIndex(index);
+        return (T) currentNode.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node currentNode = getNodeByIndex(index);
+        T currentValue = (T) currentNode.value;
+        currentNode.value = value;
+        return currentValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node deletedNode = getNodeByIndex(index);
+        unlink(deletedNode, index);
+        return (T) deletedNode.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node deletedNode = firstNode;
+        for (int i = 0; i < sizeOfList; i++) {
+            if (object == deletedNode.value || object != null && object.equals(deletedNode.value)) {
+                remove(i);
+                return true;
+            }
+            deletedNode = deletedNode.nextNode;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return sizeOfList;
     }
 
     @Override
     public boolean isEmpty() {
+        if (sizeOfList == 0) {
+            return true;
+        }
         return false;
     }
+
+    private void checkIndex(int index) {
+        if (index < sizeOfList && index >= 0) {
+            return;
+        }
+        throw new IndexOutOfBoundsException("Index out of bounds of MyLinkedList.");
+    }
+
+    private void addToStart(T value, int index) {
+        Node node = new Node(value);
+        firstNode.previousNode = node;
+        node.nextNode = firstNode;
+        firstNode = node;
+        sizeOfList += 1;
+    }
+
+    private Node getNodeByIndex(int index) {
+        if (index == 0) {
+            return firstNode;
+        }
+        int currentIndex = 0;
+        Node currentNode = firstNode;
+        while (currentIndex < index) {
+            currentNode = currentNode.nextNode;
+            currentIndex += 1;
+        }
+        return currentNode;
+    }
+
+    private void unlink(Node node, int index) {
+        Node currentNextNode = node.nextNode;
+        Node currentPrevNode = node.previousNode;
+        if (index > 0) {
+            currentPrevNode.nextNode = currentNextNode;
+        }
+        if (index == 0) {
+            firstNode = currentNextNode;
+        }
+        if (index < sizeOfList - 1) {
+            currentNextNode.previousNode = currentPrevNode;
+        }
+        if (index == sizeOfList - 1) {
+            lastNode = currentPrevNode;
+        }
+        node.previousNode = null;
+        node.nextNode = null;
```

Comment:
No need to have if for all of these cases
```suggestion
        if (node.previousNode != null) {
            node.previousNode.nextNode = node.nextNode;
        } else {
            firstNode = node.nextNode;
        }

        if (node.nextNode != null) {
            node.nextNode.previousNode = node.previousNode;
        } else {
            lastNode = node.previousNode;
        }
```